src/compojure_throttle/core.clj

```diff
@@ -72,17 +90,23 @@
    :service-compojure-throttle-enabled is false, throttling will still happen 
    to any IP not covered by :service-compojure-throttle-lax-ips.
   
-  Optionally takes a second argument which is a function used to lookup the 'token'
-  that determines whether or not the request is unique. For example a function that
-  returns a user token to limit by user id rather than ip. This function should accept
-  the request as its single argument"
-  ([finder handler]
+   Optionally takes a second argument which is a function used to lookup the 'token'
+   that determines whether or not the request is unique. For example a function that
+   returns a user token to limit by user id rather than ip. This function should accept
+   the request as its single argument.
+
+   For full generality, this also takes an optional third argument, an instance
+   of 'throttle-cache'. Create one using (make-throttle-cache ttl tokens) to
+   have a separate throttler from the global one with custom settings."
+  ([finder throttle-cache handler]
    (fn [req]
      (if (and (or (nil? (ip-lax-subnet))
                   (not (in-lax-subnet? (:remote-addr req))))
-              (throttle? (finder req)))
+              (throttle? throttle-cache (finder req)))
        {:status (prop :service-compojure-throttle-response-code)
         :body   "You have sent too many requests. Please wait before retrying."}
        (handler req))))
+  ([finder handler]
+   (throttle finder requests handler))
   ([handler]
-   (throttle by-ip handler)))
+   (throttle by-ip requests handler)))
```

Comment:
I don't know right answer. but to me I write code to adding parameter
```clojure
(defn FF
  ([a] (FF a :b))
  ([a b] (FF a b :c))
  ([a b c] (do something)))
```
this is because to make easy changing default value . for example change default from :b to bb

when you see `clojure.core` there is similar functions
```clojure
(defn read
  ([]
   (read *in*))
  ([stream]
   (read stream true nil))
  ([stream eof-error? eof-value]
   (read stream eof-error? eof-value false))
  ([stream eof-error? eof-value recursive?]
   .......
```

Reply:
Yeah you're right. For some reason I thought middlewares had a convention of putting the handler last but according to https://github.com/ring-clojure/ring/wiki/Concepts#middleware you're actually supposed to put it first. So let me see what I can do here.

Reply:
Ah.. actually, I am just suggesting
```clojure
([handler]
 (throttle by-ip handler))
```
order of parameter is always controversial. but I am not saying that now :)

Reply:
Ah I get it now. Fixed in https://github.com/kirasystems/compojure-throttle/pull/4/commits/b6ca75715fb4674ad9ad94ac6462024b29b6ff0a